From 0e852b45c0ec74ed57486202bee12fa898f3688e Mon Sep 17 00:00:00 2001
From: ckf104 <1900011634@pku.edu.cn>
Date: Wed, 30 Aug 2023 18:45:15 +0800
Subject: [PATCH] riscv: fix compiling error

Add missing header, delete keyword which is not available in 5.10 version.
---
 arch/riscv/kernel/process.c | 1 +
 arch/riscv/kernel/traps.c   | 2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/riscv/kernel/process.c b/arch/riscv/kernel/process.c
index 99b98704daf3..fe280d221b47 100644
--- a/arch/riscv/kernel/process.c
+++ b/arch/riscv/kernel/process.c
@@ -14,6 +14,7 @@
 #include <linux/tick.h>
 #include <linux/ptrace.h>
 #include <linux/uaccess.h>
+#include <linux/slab.h>
 
 #include <asm/unistd.h>
 #include <asm/processor.h>
diff --git a/arch/riscv/kernel/traps.c b/arch/riscv/kernel/traps.c
index 0bc77d5cae15..18589f683039 100644
--- a/arch/riscv/kernel/traps.c
+++ b/arch/riscv/kernel/traps.c
@@ -97,7 +97,7 @@ DO_ERROR_INFO(do_trap_insn_misaligned,
 DO_ERROR_INFO(do_trap_insn_fault,
 	SIGSEGV, SEGV_ACCERR, "instruction access fault");
 
-asmlinkage __visible __trap_section void do_trap_insn_illegal(struct pt_regs *regs)
+asmlinkage __visible void do_trap_insn_illegal(struct pt_regs *regs)
 {
 	if (user_mode(regs)) {
 		//irqentry_enter_from_user_mode(regs);
-- 
2.20.1

